web/pdf_scripting_manager.js

```diff
@@ -203,10 +205,16 @@ class PDFScriptingManager {
   }
 
   async dispatchWillPrint() {
-    return this.#scripting?.dispatchEventInSandbox({
+    if (!this.#scripting) {
+      return undefined;
+    }
+    this.#willPrintCapability = new PromiseCapability();
+    const result = await this.#scripting.dispatchEventInSandbox({
       id: "doc",
       name: "WillPrint",
     });
+    await this.#willPrintCapability.promise;
+    return result;
```

Comment:
Is there *any* risk that `dispatchWillPrint` could be called back-to-back, because if so I'm not sure that we can guarantee the event ordering (with printing intermittently not working as intended)?
Taking that in mind, I do believe that we should be even more defensive here to prevent any future issues.

```suggestion
    if (this.#willPrintCapability) {
      // Wait for a pending `dispatchWillPrint`-call to complete first.
      await this.#willPrintCapability.promise;
    }
    this.#willPrintCapability = new PromiseCapability();
    let result;

    try {
      result = await this.#scripting.dispatchEventInSandbox({
        id: "doc",
        name: "WillPrint",
      });
    } catch (ex) {
      this.#willPrintCapability = null;
      throw ex;
    }    
    await this.#willPrintCapability.promise;
    return result;
```